### PR TITLE
Fix 792

### PR DESCRIPTION
**Pull request by yzhang-gh**
On behalf of @Lemmingh.

### CODE REVIEW

**Comment by yzhang-gh** [comment]
Will do it tonight.

File: src/util.ts
Comment (yzhang-gh) [inline src/util.ts:123]: Did you check this? It is kind of special as they are CommonMark syntaxes but...

It occurred to me whether we can properly deal with a heading like `# - pretends to be a list`.

**Review by yzhang-gh — COMMENTED** [review]
👍

File: package-lock.json
Comment (Lemmingh) [inline package-lock.json]: @yzhang-gh,
Seems that you broke `package-lock.json`.
It's best regenerate it by npm.

File: package.json
Comment (Lemmingh) [inline package.json]: * You told me that [new `image-size` breaks the compatibility](https://github.com/yzhang-gh/vscode-markdown/commit/9b7fb3a5aa672847b288c6f87822dafb5b14ad6b#commitcomment-42278369).
* Due to **webpack** and **VS Code**, we have to stick to **Node.js 12** in the foreseeable future.

Thus, updating these two type definitions looks not a good idea.

**Review by Lemmingh — CHANGES_REQUESTED** [review]
Honestly, I always manipulate dependencies by `npm`, and update with extreme caution.

File: src/util.ts
Comment (Lemmingh) [reply src/util.ts:123]: They are covered by `renderInline`.

File: package-lock.json
Comment (yzhang-gh) [reply package-lock.json]: It's already regenerated (05f8d4f).

File: src/markdownEngine.ts
Comment (Lemmingh) [reply src/markdownEngine.ts:133]: https://markdown-it.github.io/markdown-it/#Renderer.renderToken

File: package.json
Comment (yzhang-gh) [reply package.json]: `image-size@0.8.0` looks fine (`0.9.1` doesn't).

> * Due to **webpack** and **VS Code**, we have to stick to **Node.js 12** in the foreseeable future.

I agree.

File: src/markdownEngine.ts
Comment (Lemmingh) [reply src/markdownEngine.ts:18]: ECMAScript private field.

I think a Markdown engine instance, as defined in `interface IMarkdownEngine`, should expose a readonly `engine` property. So I wrote `public get engine()` below.

However, the instance needs to store the engine somewhere inside. So I created private field `#engine`.

Comment (yzhang-gh) [reply src/markdownEngine.ts:18]: 👌

File: src/markdownEngine.ts
Comment (yzhang-gh) [reply src/markdownEngine.ts:133]: 👌

File: src/util.ts
Comment (yzhang-gh) [reply src/util.ts:123]: Nice!

File: src/markdownEngine.ts
Comment (yzhang-gh) [reply src/markdownEngine.ts:6]: We have another (lazy) `import` statement here

https://github.com/yzhang-gh/vscode-markdown/blob/faa18bddd35618a99b73cc093919366eff9ea432/src/markdownEngine.ts#L49

although I don't think it improves performance much. Many features rely on it. Probably we also change this?

Comment (Lemmingh) [reply src/markdownEngine.ts:6]: To ... what? Could you please explain?

I don't know whether webpack will handle this case in some special way.

Also see my note:
https://github.com/users/Lemmingh/projects/2#card-44982812

Comment (yzhang-gh) [reply src/markdownEngine.ts:6]: (I mean `MarkdownIt` is already imported so we don't need to import it again.)

Comment (yzhang-gh) [reply src/markdownEngine.ts:6]: > To ... what? Could you please explain?

The activation time

Comment (Lemmingh) [reply src/markdownEngine.ts:6]: I see.

I agree to change `(await import('markdown-it'))` to `new MarkdownIt`.

Comment (yzhang-gh) [reply src/markdownEngine.ts:6]: Let's do it and :shipit:

**Comment by yzhang-gh** [comment]
Just merge this in when you are ready 😉

**Review by Lemmingh — APPROVED** [review]

**Comment by yzhang-gh** [comment]
Help yourself 👌

**Comment by yzhang-gh** [comment]
🍻

**Comment by Lemmingh** [comment]
😂

BTW, could you please disable the **merge commits** option?
**Settings** > **Options** > **Merge button**, uncheck **Allow merge commits**.

Such a long drop down menu is a visual clutter to me.

**Comment by yzhang-gh** [comment]
Ah, didn't know about this option.

**Pull request merged**
